assets/src/scripts/components/hydrograph/time-series-graph.spec.js

```diff
@@ -0,0 +1,388 @@
+                            }, {
+                                dateTime: 1514930400000,
+                                value: null,
+                                qualifiers: ['P', 'FLD']
+                            }]
+                        }
+                    }
+                },
+                timeSeriesState: {
+                    showSeries: {
+                        current: true,
+                        compare: true,
+                        median: true
+                    },
+                    currentVariableID: '45807197',
+                    currentDateRange: 'P7D',
+                    currentMethodID: 'method1',
+                    loadingTSKeys: []
+                },
+                ui: {
+                    windowWidth: 400,
+                    width: 400
+                }
+
+            });
+
+            attachToNode(store, graphNode, {siteno: '123456788'});
+        });
+
+        it('should render the correct number of svg nodes', () => {
+            // one main hydrograph, legend and two sparklines
+            expect(selectAll('svg').size()).toBe(4);
+        });
+
+        it('should have a title div', () => {
+            const titleDiv = selectAll('.time-series-graph-title');
+            expect(titleDiv.size()).toBe(1);
+            expect(titleDiv.text()).toEqual('Test title for 00060, method description');
+        });
+
+        it('should have a defs node', () => {
+            expect(selectAll('defs').size()).toBe(1);
+            expect(selectAll('defs mask').size()).toBe(1);
+            expect(selectAll('defs pattern').size()).toBe(2);
+        });
+
+        it('should render time series data as a line', () => {
+            // There should be one segment per time-series. Each is a single
+            // point, so should be a circle.
+            expect(selectAll('.hydrograph-svg .line-segment').size()).toBe(2);
+        });
+
+        it('should render a rectangle for masked data', () => {
+            expect(selectAll('.hydrograph-svg g.current-mask-group').size()).toBe(1);
+        });
+
+        it('should have a point for the median stat data with a label', () => {
+            expect(selectAll('#median-points path').size()).toBe(1);
+            expect(selectAll('#median-points text').size()).toBe(0);
+        });
+
+        it('should have tooltips for the select series table', () => {
+            // one for each of the two parameters
+            expect(selectAll('table .tooltip-item').size()).toBe(2);
+        });
+
+        it('should not have tooltips for the select series table when the screen is large', () => {
+            store.dispatch(Actions.resizeUI(800, 800));
+            expect(selectAll('table .tooltip-table').size()).toBe(0);
+        });
+    });
+
+    //TODO: Consider adding a test which checks that the y axis is rescaled by
+    // examining the contents of the text labels.
+
+    describe('legends should render', () => {
```

Comment:
This doesn't belong here at all because the legend code is not part of the time-series-graph rendering.